plugin/gcp/storageObject.go

```diff
@@ -45,6 +45,16 @@ func (s *storageObject) Read(ctx context.Context, size int64, offset int64) ([]b
 	return ioutil.ReadAll(rdr)
 }
 
+func (s *storageObject) Write(ctx context.Context, p []byte) error {
+	wr := s.ObjectHandle.NewWriter(ctx)
+	if _, err := wr.Write(p); err != nil {
+		wr.Close()
+		return err
+	}
+
+	return wr.Close()
```

Comment:
If `wr.Close()` fails, does that mean the data isn't written (whereby returning a Write error makes sense)? If not, then this should probably be logged to the activity journal or something similar.

Reply:
Yes, it means the data wasn't successfully written.